OneMore/Commands/File/Markdown/MarkdownConverter.cs

```diff
@@ -52,6 +52,30 @@ public void RewriteHeadings()
 			{
 				RewriteHeadings(outline.Descendants(ns + "OE"));
 			}
+
+			// added header spacing specific to markdown
+			var quickstyles = page.Root.Elements(ns + "QuickStyleDef");
+			foreach (var quickstyle in quickstyles)
+			{
+				var name = quickstyle.Attribute("name").Value;
+				if (name.Equals("h1") || name.Equals("h2"))
+				{
+					replaceAtributes(quickstyle, spaceBefore: 0.8, spaceAfter: 0.5);
+				}
+				else
+				if (name.Equals("h3") || name.Equals("h4"))
+				{
+					replaceAtributes(quickstyle, spaceBefore: 0.3, spaceAfter: 0.3);
+				}
+			}
+			void replaceAtributes(XElement quickstyle, double spaceBefore, double spaceAfter)
+			{
+				quickstyle.Attributes().Where(a => a.Name == "spaceBefore").Remove();
```

Comment:
No need to remove attributes before setting them. The SetAttributeValue extension will set them correctly. There can be only one instance of each attribute in each span element.

Reply:
very good - reworked accordingly.